src/services/similarity.js

```diff
@@ -23,15 +23,28 @@ async function postToSimilarity(body) {
         body: JSON.stringify(body),
       },
     );
-    auditLogger.info(`${namespace} Similarity API response status: ${response.status}, body: ${JSON.stringify(response.body)}`);
+
+    // Check if response is OK before parsing JSON
+    if (!response.ok) {
+      const errorText = await response.text();
+      auditLogger.error(`${namespace} Similarity API error response: ${errorText.substring(0, 500)}`);
+      throw new Error(`API returned status ${response.status}: ${errorText.substring(0, 100)}`);
+    }
 
     return await response.json();
   } catch (error) {
-    auditLogger.error(
-      `${namespace} Similarity API response failure: ${error.message}`,
-      { error },
-    );
-    throw new Error(error);
+    if (error.name === 'SyntaxError') {
+      auditLogger.error(
+        `${namespace} Failed to parse JSON response from Similarity API: ${error.message}`,
+        { error },
+      );
+    } else {
+      auditLogger.error(
+        `${namespace} Similarity API response failure: ${error.message}`,
+        { error },
+      );
+    }
+    throw error;
```

Comment:
do we need to both check response.ok and check specifically for syntax errors? What I mean is: we'd get valid JSON as long as the response is ok, wouldn't we?

Reply:
to be more clear, I am confused only about the purpose of the if statement starting on line 36

Reply:
right - probably not necessary. I will remove this check